backend-shared/config/db/util/utils_test.go

```diff
@@ -1,12 +1,881 @@
+// Check required KubernetesToDBResourceMapping is present in Db.
+// Since we don't know "kubernetes_resource_uid" to fetch object using GetDBResourceMappingForKubernetesResource function
+// we will have to fetch all entries from table and check one by one if expected entry is present in table.
+func findKubernetesToDBResourceMappingInTable(ctx context.Context, dbQueries db.AllDatabaseQueries, dBRelationKey string, dBRelationType string) (db.KubernetesToDBResourceMapping, bool, error) {
+
+	foundMapping := false
+	var mappingResult db.KubernetesToDBResourceMapping
+	var kubernetesToDBResourceMappings []db.KubernetesToDBResourceMapping
+
+	err := dbQueries.UnsafeListAllKubernetesResourceToDBResourceMapping(ctx, &kubernetesToDBResourceMappings)
+	if err != nil {
+		return mappingResult, foundMapping, err
+	}
+
+	for i := range kubernetesToDBResourceMappings {
+		item := kubernetesToDBResourceMappings[i]
+		if item.DBRelationType == dBRelationType &&
+			item.DBRelationKey == dBRelationKey {
+
+			// The entry we are looking for is present in table
+			foundMapping = true
+			mappingResult = item
+			break
+		}
+	}
+	return mappingResult, foundMapping, nil
+}
+
+// initialize object required prior to tests
+func initialSetUp() (context.Context, db.AllDatabaseQueries, logr.Logger, types.UID, error) {
+	dbQueries, err := db.NewUnsafePostgresDBQueries(false, false)
+	if err != nil {
+		return nil, nil, nil, "", err
+	}
+
+	ctx := context.Background()
+	log := logger.FromContext(ctx)
+
+	return ctx, dbQueries, log, uuid.NewUUID(), err
+}
+
+var _ = Describe("Test utility functions.", func() {
+
+	Context("Testing for GetOrCreateManagedEnvironmentByNamespaceUID function.", func() {
+
+		It("Should create new managedEnvironment and other resources, if called second time then it should return existing resources.", func() {
+			ctx, dbQueries, log, workSpaceUid, err := initialSetUp()
+			Expect(err).To(BeNil())
+
+			defer dbQueries.CloseDatabase()
+
+			workspace := v1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-user",
+					UID:       workSpaceUid,
+					Namespace: "test-namespace",
+				},
+				Spec: v1.NamespaceSpec{},
+			}
+
+			// ----------------------------------------------------------------------------
+			By("It should create new ManagedEnvironment for given namespace.")
+			// ----------------------------------------------------------------------------
+
+			managedEnvironment, isNew, err := GetOrCreateManagedEnvironmentByNamespaceUID(ctx, workspace, dbQueries, log)
+			Expect(err).To(BeNil())
+			Expect(isNew).To(BeTrue())
+
+			// ----------------------------------------------------------------------------
+			By("Verify that other database entries are also created.")
+			// ----------------------------------------------------------------------------
+
+			// Check KubernetesToDBResourceMapping resource
+			kubernetesToDBResourceMapping := db.KubernetesToDBResourceMapping{
+				KubernetesResourceType: "Namespace",
```

Comment:
You can use `db.K8sToDBMapping_Namespace` for this

Reply:
Yes, I missed here.